client/mm/libxc/mexc.go

```
@@ -0,0 +1,3248 @@
+	// This ensures Markets() will have the tradeable list when called
+	if err := m.getDefaultSymbols(ctx); err != nil {
+		m.log.Errorf("Failed to fetch API-tradeable symbols: %v", err)
+		// Don't fail - Markets() will use optimistic default (assume all tradeable)
+	}
+
+	return nil
+}
+
+func (m *mexc) symbolExists(sym string) bool {
+	m.symbolsMtx.RLock()
+	_, ok := m.symbols[strings.ToUpper(sym)]
+	m.symbolsMtx.RUnlock()
+	return ok
+}
+
+// getSymbolMeta returns the symbol metadata for a given symbol.
+func (m *mexc) getSymbolMeta(symbol string) *mxctypes.Symbol {
+	m.symbolMetaMtx.RLock()
+	defer m.symbolMetaMtx.RUnlock()
+	return m.symbolMeta[strings.ToUpper(symbol)]
+}
+
+// getFilterValue returns the value of a specific filter type for a symbol.
+func (m *mexc) getFilterValue(symbol, filterType string) string {
+	meta := m.getSymbolMeta(symbol)
+	if meta == nil {
+		return ""
+	}
+	for _, filter := range meta.Filters {
+		if filter.FilterType == filterType {
+			switch filterType {
+			case "PRICE_FILTER":
+				return filter.TickSize
+			case "LOT_SIZE":
+				return filter.StepSize
+			case "MIN_NOTIONAL":
+				return filter.MinNotional
+			case "MIN_QTY":
+				return filter.MinQty
+			case "MAX_QTY":
+				return filter.MaxQty
+			}
+		}
+	}
+	return ""
+}
+
+// snapPrice snaps a price to the tick size for a symbol.
+func (m *mexc) snapPrice(symbol string, price float64) float64 {
+	tickSizeStr := m.getFilterValue(symbol, "PRICE_FILTER")
+	if tickSizeStr == "" {
+		return price
+	}
+	tickSize, err := strconv.ParseFloat(tickSizeStr, 64)
+	if err != nil {
+		return price
+	}
+	return float64(int64(price/tickSize+0.5)) * tickSize
+}
+
+// snapQuantity snaps a quantity to the step size for a symbol.
+func (m *mexc) snapQuantity(symbol string, qty float64) float64 {
+	stepSizeStr := m.getFilterValue(symbol, "LOT_SIZE")
+	if stepSizeStr == "" {
+		return qty
+	}
+	stepSize, err := strconv.ParseFloat(stepSizeStr, 64)
+	if err != nil {
+		return qty
+	}
```

I don't think it is good to ignore the error. It cannot work correctly if there is error.